backend/apps/packages/views.py

```
@@ -218,3 +221,109 @@ def post(self, request):
+        residence = getattr(request, "residence", None)
+        if not residence:
+            raise ValidationError({"detail": ERROR_NO_RESIDENCE})
+
+        from_str = (request.query_params.get("from") or "").strip()
+        to_str = (request.query_params.get("to") or "").strip()
+
+        try:
+            from_date = date.fromisoformat(from_str) if from_str else date.today() - timedelta(days=29)
+            to_date = date.fromisoformat(to_str) if to_str else date.today()
+        except ValueError:
+            raise ValidationError({"detail": "Formato de fecha inválido. Usa YYYY-MM-DD."})
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)